### ISSUE TITLE
Worse metric than paper mentioned

### ISSUE THREAD

**Issue by BuJiaLaDiii** [PeiwenSun2000/Both-Ears-Wide-Open#7]
Recently I try to test **BEWO_nl** model on **Audiocaps test set**, with code as below. I use code on **'./models/simple_generation.py'**. 
But I get worse metric than paper mentioned.

This code I not change the generation code:
`
def generate_audio(prompt, model, device, output_dir, sample_idx, resampler):
    conditioning = [{"prompt": prompt, "seconds_start": 0, "seconds_total": 10}]

    audio = generate_diffusion_cond(
        model,
        conditioning=conditioning,
        negative_conditioning=None,
        steps=250,
        cfg_scale=7.5,
        batch_size=1,
        sample_size=int(44100 * 10),
        sample_rate=44100,
        seed=-1,
        device=device,
        sampler_type="dpmpp-3m-sde",
        sigma_min=0.03,
        sigma_max=1000,
        init_audio=None,
        init_noise_level=1.0,
        mask_args=None,
        callback=None,
        scale_phi=0
    )

    audio_16k = resampler(audio.squeeze(0).cpu())
    audio_16k_filename = f"test_sample_{sample_idx}.wav"
    torchaudio.save(output_dir / audio_16k_filename, audio_16k, sample_rate=16000)

    return {
        "sample_index": sample_idx,
        "prompt": prompt,
        "audio_filename": audio_16k_filename,
    }

if __name__ == "__main__":
    output_dir = Path("/workspace/BothEars/audiocaps_test")
    output_dir.mkdir(parents=True, exist_ok=True)

    resampler = torchaudio.transforms.Resample(orig_freq=44100, new_freq=16000)

    model_config_path = "/workspace/BothEars/models/bewo_config/model_config_sim.json"
    model_ckpt_path = "/workspace/BothEars/BEWO_nl.ckpt"

    with open(model_config_path) as f:
        model_config = json.load(f)

    model, model_config = load_model(
        model_config=model_config,
        model_ckpt_path=model_ckpt_path,
        device='cuda:1'
    )

    # 准备数据集
    test_ds = AudioCaps(data_type="test")
    test_dataloader = torch.utils.data.DataLoader(
        test_ds,
        batch_size=1,
        shuffle=False,
        num_workers=4
    )

    # 初始化结果数据结构
    generated_data = {
        "epoch": "test",
        "samples": []
    }

    # 生成音频
    for sample_idx, batch in enumerate(tqdm(test_dataloader, desc="Generating samples")):
        prompt = batch['prompt'][0]

        sample_info = generate_audio(
            prompt=prompt,
            model=model,
            device='cuda:1',
            output_dir=output_dir,
            sample_idx=sample_idx,
            resampler=resampler
        )

        generated_data["samples"].append(sample_info)

    metadata_path = output_dir / "test_metadata.json"
    with open(metadata_path, 'w', encoding='utf-8') as f:
        json.dump(generated_data, f, ensure_ascii=False, indent=4)

    print(f"Generated {len(generated_data['samples'])} samples")
    print(f"Metadata saved to {metadata_path}")`

And result as blow:
`{'frechet_distance': 89.97694638372688, 'frechet_audio_distance': 15.051326558644718, 'kullback_leibler_divergence_sigmoid': 11.803584098815918, 'kullback_leibler_divergence_softmax': 4.642857074737549, 'lsd': nan, 'psnr': nan, 'ssim': nan, 'inception_score_mean': 3.2329116491147487, 'inception_score_std': 0.47343594734223854, 'kernel_inception_distance_mean': nan, 'kernel_inception_distance_std': nan}`

**Comment by PeiwenSun2000** [comment]
There might be the following reasons.

1. As we mention in our paper, we adopt evaluation in [MAA](https://github.com/Text-to-Audio/Make-An-Audio). They introduce a subset of Audiocaps to avoid unnecessary evaluation.
2. It is common practice to sample multiple times and select the best results, like [AudioLDM](https://github.com/haoheliu/AudioLDM)

However, the above tips are only for boosting the minor performance gap. The results you have shown are A LOT WORSE. frechet_distance is 89.97694638372688, which means probably the result is somehow noise.

During my experiments, if I use the newer version of the stable-audio-tools, the results are noisy. So, another suggestion is

1. please uninstall stable-audio-tools and stick to the stable version provided in [model](https://github.com/PeiwenSun2000/Both-Ears-Wide-Open/tree/main/models/stable_audio_tools)

**Comment by BuJiaLaDiii** [comment]
> There might be the following reasons.
> 
> 1. As we mention in our paper, we adopt evaluation in [MAA](https://github.com/Text-to-Audio/Make-An-Audio). They introduce a subset of Audiocaps to avoid unnecessary evaluation.
> 2. It is common practice to sample multiple times and select the best results, like [AudioLDM](https://github.com/haoheliu/AudioLDM)
> 
> However, the above tips are only for boosting the minor performance gap. The results you have shown are A LOT WORSE. frechet_distance is 89.97694638372688, which means probably the result is somehow noise.
> 
> During my experiments, if I use the newer version of the stable-audio-tools, the results are noisy. So, another suggestion is
> 
> 1. please uninstall stable-audio-tools and stick to the stable version provided in [model](https://github.com/PeiwenSun2000/Both-Ears-Wide-Open/tree/main/models/stable_audio_tools)

**Comment by PeiwenSun2000** [comment]
Oh, sorry for the confusion. My proposed evaluation is only about ITD, which is related to Table 2 on our BEWO test set. 

However, the evaluation code for 1-channel T2A on the AudioCaps test should be the same as the previous papers... After all, it is not our contribution.

If you have other problems, feel free to mail me!

**Comment by BuJiaLaDiii** [comment]
Same results , I try calculating the CLAP_score with code as below. But CLAP_score is around 0.1~0.05. CFG is 7.5

`import os
import json
import numpy as np
import librosa
import torch
import laion_clap
from tqdm import tqdm
import pyloudnorm as pyln


def int16_to_float32(x):
    return (x / 32767.0).astype(np.float32)


def float32_to_int16(x):
    x = np.clip(x, a_min=-1., a_max=1.)
    return (x * 32767.).astype(np.int16)


model = laion_clap.CLAP_Module(enable_fusion=True, device='cpu')
model.load_ckpt('/workspace/dky_evaluation/both ear/evaluations/checkpoints/630k-audioset-fusion-best.pt')

base_path = '/workspace/BothEars/audiocaps_test_1'
test_metadata_path = f'{base_path}/test_metadata.json'


with open(test_metadata_path, 'r') as f:
    data = json.load(f)
    samples = data.get('samples', [])

id2text = {}
for sample in samples:
    caption = sample.get('prompt')
    audio_filename = sample.get('audio_filename')
    id2text[audio_filename] = caption

**Comment by PeiwenSun2000** [comment]
CLAP_score is around 0.1-0.05. It is too low to be normal. I think if you test on noise, the CLAP would be around 0.1~0.05...

Would you mind leaving your email here? I will send you our raw test results for your reference.

Another thing might help. Can you change the code to the Stable-audio-open model and see if the results are normal? It helps narrow down the scope of the problem.

I will help you with everything I can. It also helps to build the impact of this paper. Just leave your email if you need it.

**Comment by BuJiaLaDiii** [comment]
daikunyan@shu.edu.cn
